src/ray/object_manager/pull_manager.cc

```diff
@@ -89,43 +87,51 @@ void PullManager::TryToMakeObjectLocal(const ObjectID &object_id) {
   if (it == object_pull_requests_.end()) {
     return;
   }
-
   auto &request = it->second;
+  if (request.next_pull_time > get_time_()) {
+    return;
+  }
+
+  bool did_pull = false;
```

Comment:
Nit, but I think it's more readable to just update the retry timer in the same blocks as the PullFromRandomLocation calls instead of sharing this bool (the two branches that use it are independent anyway).